packages/grafana-ui/src/components/PieChart/PieChart.tsx

```diff
@@ -187,31 +224,15 @@ export const PieChartSvg: FC<SvgProps> = ({
           >
             {(pie) => {
               return pie.arcs.map((arc) => {
-                return (
-                  <g
-                    key={arc.data.title}
-                    className={styles.svgArg}
-                    onMouseMove={(event) => onMouseMoveOverArc(event, arc.data)}
-                    onMouseOut={hideTooltip}
-                  >
-                    <path
-                      d={pie.path({ ...arc })!}
-                      fill={useGradients ? getGradientColor(arc.data.color ?? FALLBACK_COLOR) : arc.data.color}
-                      stroke={theme.colors.panelBg}
-                      strokeWidth={1}
-                    />
-                    {showLabel && (
-                      <PieLabel
-                        arc={arc}
-                        outerRadius={layout.outerRadius}
-                        innerRadius={layout.innerRadius}
-                        displayLabels={displayLabels}
-                        total={total}
-                        color={theme.colors.text}
-                      />
-                    )}
-                  </g>
-                );
+                if (arc.data.hasLinks && arc.data.getLinks) {
+                  return (
+                    <DataLinksContextMenu key={arc.index} links={arc.data.getLinks}>
+                      {(api) => getPieSlice(arc, pie, api.openMenu)}
+                    </DataLinksContextMenu>
+                  );
```

Comment:
In a situation when there is only a single data link I would expect no context menu and only the arc being clickable. Ref: https://github.com/grafana/grafana/issues/26805

Reply:
@dprokop think the change to https://github.com/grafana/grafana/pull/31692 I think this will be a bit tricky to use here as DataLinksContextMenu renders anchor , and I do not think an anchor is allowed inside an svg.

Reply:
Shit, I think they are allowed but require `xlink:href` attribute instead of `href`